### ISSUE TITLE
New GitHub issue and PR template

### ISSUE THREAD

**Issue by antoniogamiz** [Raku/doc#2904]
## The problem

If you go `.github` directory you will see this warning:

![1](https://user-images.githubusercontent.com/22995015/61278372-5f84c780-a7b4-11e9-80ac-77e996904302.png)

## Suggestions

Update the template to match the new format.

**Comment by antoniogamiz** [comment]
You can even specify a template for bugs => https://help.github.com/en/articles/creating-issue-templates-for-your-repository

**Comment by antoniogamiz** [comment]
This is related to https://github.com/perl6/problem-solving/issues/65.

**Comment by treyharris** [comment]
@JJ  Oops, I assigned myself this because I’ve done it before elsewhere, but it looks like the roles for the Perl6 org have changed since I last did repo housekeeping and I don’t have access to repo settings anymore.

Could you change my role, please?

**Comment by AlexDaniel** [comment]
@treyharris done. I added you as an admin to this repo specifically, and I hope to review and tweak all permissions a bit later so that they're more consistent and useful. Ping me if you need more.

**Comment by treyharris** [comment]
Note pull request #2912 I’ve just pushed up. I’d appreciate any thoughts about them (@JJ from you especially) either here or as a code review comment.

Note that I’ve pushed up a couple fixes since my initial push, and that the comment I made about new PR template was in error—I didn’t notice that the “old” PR template location is a degenerate case of the “new” one, and the degenerate case (a single template file for the entire repo) is the one we probably want to use.

(Although it _does_ appear like we could treat `/doc/*`-only PR’s differently from ones that touch other directories, that only seems superficially useful since I imagine most people create PR’s using desktop tools that aren’t using the nonstandard GitHub query parameters, so won’t have a chance to see the PR templates regardless.)

**Comment by JJ** [comment]
I think that a couple of templates, one for bugs and another for enhancements or clarifications would be more than enough; even just converting the template to the new format would be enough. Don't try to fix something that's not broken, and fix only what's broken, in this case the format of the specific dir and/or template.

**Comment by treyharris** [comment]
If literally all you wanted was to get rid of the warning—I’ve gotten rid of it already. But this ticket said, “new issue templates”, by which I thought you meant you wanted... new issue templates. If the intent was simply to get rid of the warning message and anything beyond that was “fixing what wasn’t broken”, that certainly wasn’t expressed earlier in this ticket.

The original template text was vague because you could have just one template, so it needed to apply to anyone (with the understanding that those who are frequent contributors would probably just ignore the template entirely). The new template format allows for multiple templates, so I looked through the issue history and label assignments and created ones that covered the most frequent cases for non-contributors, and threw in two additional ones for contributors (a catch-all one and an operations one).

I’m honestly a bit annoyed that _you_ are annoyed that I put some thought and work into this based on actual observed examples of wheel-spinning and duplication of effort. Wordsmithing is fine, but the suggestion here seems to be that my proposed solution is worse than the current status. I took on the issue based on how it was requested. If the request was “don’t change anything, just get rid of the warning”, I very much would have appreciated that being mentioned before giving me the additional access signaling to me that some effort was desired.

But feel free to close this ticket and reject the pull request if this was all a big misunderstanding.

**Comment by treyharris** [comment]
@JJ  — Your comment on the pull request:

> I'm commenting on the #2904 issue. Can we please discuss what's the issue there while we leave this for discussing the solution? TL;DR: I think what we need only is to update issue and PRs templates. New issue templates should be proposed, and discussed, separately.

may shed some light on the misunderstanding here. Let’s concentrate on “what we need only is to update... PRs templates”.

You can’t create a commit that both “updates” pull request templates _and_ leaves their contents alone, because in the case of PR’s the degenerate case of the new format _is_ the old format. A commit with changes implies either wording changes, functionality changes, or both. So if you could describe what you were expecting in a “new” PR template, perhaps I’d better understand what you were asking for in the first place.

**Comment by JJ** [comment]
@treyharris I didn't create the original post, but it just says "update the template to match the new format". To close this issue, that's all that is needed.

**Comment by treyharris** [comment]
> @treyharris I didn't create the original post, but it just says "update the template to match the new format". To close this issue, that's all that is needed.

But as I wrote above, if no wording or functionality changes are desired, there _is_ no update to do. If you're just saying you want a commit for the sake of a commit, I could make a functionally null-op whitespace change or filename change or something, but I thought there was a desire for more than that.

This is an XY problem, it seems: I read "create new templates" as being the X problem, meaning "create new templates to take advantage of the new functionality" was the obvious solution. That's what I volunteered to do.

But actually, the X request was, "get rid of this tooltip warning" and needing new templates was a mistakenly-informed Y solution? Have I got that right? If so, I'll edit the title of this issue to "Remove template warning" and close the issue, since that's already done.
